app/talk/comment-box.jsx

```diff
@@ -0,0 +1,204 @@
+  EMAIL_NOT_CONFIRMED: 'Your account email address is not yet confirmed. Please check your email for confirmation instructions, or visit https://www.zooniverse.org/settings/email to request new confirmation email.'
+};
+
+function getErrorMessage(error, user) {
+  if (!error) return '';
+
+  if (
+    error.message?.match?.(/not allowed to create this Comment/i)
+    || error.message?.match?.(/not allowed to create this Discussion/i)
+    || error.message?.match?.(/You must confirm your account/i)
+  ) {
+    if (!user?.confirmed_at) {
+      return ERROR_MESSAGES.EMAIL_NOT_CONFIRMED;
+    }
+  } else if (
+    error.message?.match?.(/You are banned/i)
+  ) {
+    return ERROR_MESSAGES.BANNED;
+  }
+
+  return error.message || 'Unknown error';
+}
+
+const CommentBox = createReactClass({
+  displayName: 'Commentbox',
+  mixins: [ToggleChildren, Feedback],
+
+  propTypes: {
+    submit: PropTypes.string,
+    user: PropTypes.object,
+    header: PropTypes.string,
+    placeholder: PropTypes.string,
+    submitFeedback: PropTypes.string,
+    onSubmitComment: PropTypes.func, // called on submit and passed (e, textarea-content, subject), expected to return something thenable
+    onCancelClick: PropTypes.func
+  }, // adds cancel button and calls callback on click if supplied
+
+  getDefaultProps() {
+    return {
+      submit: "Submit",
+      header: "Add to the discussion +",
+      placeholder: "Type your comment here",
+      submitFeedback: "Comment Successfully Submitted",
+      content: '',
+      subject: null,
+      user: null,
+      logSubmit: false
+    };
+  }, // if true, geordi logging will be made through onSubmitComment
+
+  getInitialState() {
+    return {
+      subject: this.props.subject,
+      content: this.props.content,
+      reply: null,
+      loading: false,
+      error: ''
+    };
+  },
+
+  contextTypes: {
+    geordi: PropTypes.object
+  },
+
+  logPageClick(clicked, button) {
+    return this.context?.geordi?.logEvent({
+      type: clicked,
+      data: button
+    });
+  },
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.reply !== this.props.reply) {
+      return this.setState({reply: nextProps.reply});
+    }
+  },
+
+  onSubmitComment(e) {
+    if (this.props.logSubmit === true) {
+      this.logPageClick('add-comment', this.props.submit);
+    }
+    e.preventDefault();
+    const textareaValue = this.state.content;
+    if (this.props.validationCheck?.(textareaValue)) { return; }
+    this.setState({loading: true});
+
+    return this.props.onSubmitComment?.(e, textareaValue, this.state.subject, this.state.reply)
+      .then(() => {
+        this.hideChildren();
+        this.setState({subject: null, content: '', error: '', loading: false});
+        return this.setFeedback(this.props.submitFeedback);
+    }).catch(e => {
+        const errorMessage = getErrorMessage(e, this.props.user);
+        return this.setState({error: errorMessage, loading: false});
```

Comment:
Dev Notes 2/2: this is one of two additions/modifications to the CommentBox code.

Here, we use the new `getErrorMessage()` to determine the appropriate error message for volunteers.

The rest of the changes is just a result of decaffeinating the CoffeeScript.